### PR TITLE
Enable leader election for controller manager

### PR DESCRIPTION
**Pull request by khos2ow**
Add leader election for controller manager to base template.

### CODE REVIEW

File: cmd/provider/main.go [hunk 1]
```diff
@@ -52,7 +53,11 @@ func main() {
 	cfg, err := ctrl.GetConfig()
 	kingpin.FatalIfError(err, "Cannot get API server rest config")
 
-	mgr, err := ctrl.NewManager(cfg, ctrl.Options{SyncPeriod: syncPeriod})
+	mgr, err := ctrl.NewManager(cfg, ctrl.Options{
+		LeaderElection:   *leaderElection,
+		LeaderElectionID: "crossplane-leader-election-provider-<NAME>",
```
Comment (khos2ow) [inline cmd/provider/main.go:58]: I put `<NAME>` here which needs to get replaced with the actual name of the provider when a new project was created from this template. Is this enough and self-explanatory?

File: cmd/provider/main.go [hunk 2]
```diff
@@ -32,9 +32,10 @@ import (
 
 func main() {
 	var (
-		app        = kingpin.New(filepath.Base(os.Args[0]), "Template support for Crossplane.").DefaultEnvars()
-		debug      = app.Flag("debug", "Run with debug logging.").Short('d').Bool()
-		syncPeriod = app.Flag("sync", "Controller manager sync period such as 300ms, 1.5h, or 2h45m").Short('s').Default("1h").Duration()
+		app            = kingpin.New(filepath.Base(os.Args[0]), "Template support for Crossplane.").DefaultEnvars()
+		debug          = app.Flag("debug", "Run with debug logging.").Short('d').Bool()
+		syncPeriod     = app.Flag("sync", "Controller manager sync period such as 300ms, 1.5h, or 2h45m").Short('s').Default("1h").Duration()
+		leaderElection = app.Flag("leader-election", "Use leader election for the conroller manager.").Short('l').Default("false").OverrideDefaultFromEnvar("LEADER_ELECTION").Bool()
```
Comment (negz) [inline cmd/provider/main.go:38]:
```suggestion
		leaderElection = app.Flag("leader-election", "Use leader election for the controller manager.").Short('l').Default("false").OverrideDefaultFromEnvar("LEADER_ELECTION").Bool()
```

File: cmd/provider/main.go (hunk 1 above)
Comment (negz) [inline cmd/provider/main.go:58]:
```suggestion
		LeaderElectionID: "crossplane-leader-election-provider-template",
```

**Review by negz — COMMENTED** [review]
Thanks @khos2ow! I think using `template` is fine as a placeholder - we already do that in a few places.

**Review by negz — APPROVED** [review]

File: cmd/provider/main.go (hunk 2 above)
Comment (khos2ow) [reply cmd/provider/main.go]: Nice catch!

**Pull request merged**